gluon-mesh-vpn-wireguard/files/lib/gluon/gluon-mesh-wireguard/checkuplink

```diff
@@ -52,63 +59,36 @@ if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wiregu
-			SPEED2=$(curl https://$URL/speed.php -k)                                           
-                	if [ $SPEED2 -gt $SPEED1 ]              # Wenn aktueller Server hoehren Wert hat,
-                        		then AUSWAHL=$I                 # Servernummer speichern                 
-                        	SPEED1=$SPEED2                                                           
-                	fi                                                                               
-        	done                                                                                     
-        	URL=$(uci get wireguard.peer_$AUSWAHL.endpoint | cut -d':' -f1)                          
+		modprobe wireguard                                                       
+
+                NUMBER_OF_PEERS=$(uci show wireguard | egrep -e peer_[0-9].endpoint | wc -l)
+                PEER=$(awk 'BEGIN{srand();print int(rand()*($NUMBER_OF_PEERS)+1) }')
                                                                                                  
 		#Die alten peers loeschen
 		wg set $INTERFACE peer  $(wg |  awk 'BEGIN {RS=""} /endpoint/ {print $2}') remove
-                batctl if del gre
-                ip link delete dev gre
+                batctl if del vxlan-wan
+                ip link delete dev vxlan-wan
                 ip link delete dev $INTERFACE
                 PUBLICKEY=$(uci get wireguard.wireguard.privatekey | wg pubkey)
                 echo $(uci get wireguard.wireguard.privatekey) > /tmp/wgpriv
 
                 # Public Key zum vorher ausgesuchten Server hochladen
-                IP=$(curl -s -k --data-urlencode "pubkey=$PUBLICKEY" https://$URL/wireguard.php)
+                #curl -s -k --data-urlencode "pubkey=$PUBLICKEY" https://$URL/wireguard.php
 		ip link add dev $INTERFACE type wireguard
-                ip address add dev $INTERFACE $IP/16
                 wg set $INTERFACE private-key /tmp/wgpriv
                 sleep 5
 
-                wg set $INTERFACE peer $(uci get wireguard.peer_$AUSWAHL.publickey) persistent-keepalive 25 allowed-ips $(uci get wireguard.wireguard.gretapip)/32 endpoint $(uci get wireguard.peer_$AUSWAHL.endpoint)
+                wg set $INTERFACE peer $(uci get wireguard.peer_$PEER.publickey) persistent-keepalive 25 allowed-ips fe80::/10,$(uci get wireguard.wireguard.mcastgroup)/128 endpoint $(uci get wireguard.peer_$PEER.endpoint)
                 ip link set up dev $INTERFACE
+                ip address add "$(interface_linklocal "$INTERFACE")"/64 dev $INTERFACE
 
                 sleep 5
-                ip link add gre type gretap local $IP remote $(uci get wireguard.wireguard.gretapip)
-                ip link set up dev gre
+                ip link add vxlan-wan type vxlan id "$(lua -e 'print(tonumber(require("gluon.util").domain_seed_bytes("gluon-mesh-vxlan", 3), 16))')" group $(uci get wireguard.wireguard.mcastgroup) dstport 4789 dev $INTERFACE
```

Comment:
So one Wireguard interface on the Gateways and from there fan-out via VXLAN for each domain, okay.

Reply:
Actually you can also add more wireguard interfaces. Our plan was one per domain. But to make it even harder to bridge different domains, we also add the domain seed here.